### ISSUE TITLE
Portable Mode

### ISSUE THREAD

**Issue by DachsbauTV** [bitfocus/companion#1465]
**Describe the feature**
Unsure if it this would actually be possible, but I would love a portable mode of Companion, like OBS has.

**Usecases**
It would allow me to have Companion installed on a cloud drive which synchronises across my multiple streaming PCs. This would make updating all of my systems a way easier process.

**Comment by Julusian** [comment]
Would you want the config to also be stored with the executable in the drive?

**Comment by DachsbauTV** [comment]
Yes, if possible it would be ideal to have all necessary files to run companion in one folder.

**Comment by fdebeer** [comment]
> Yes, if possible it would be ideal to have all necessary files to run companion in one folder.

or run it from usb-stick as standalone program. easy to change from one pc to another. (dont know if this is possible on mac or linux)

**Comment by DachsbauTV** [comment]
After loosing my .db file for a second time today in the last couple of weeks an idea came into my mind,, which would be easy to implement and would at least make the companion setup cloud based and safe:

Would it be possible to define where the db files oder the whole %APPDATA%/companion folder is stored?

This way one could place it in a cloud drive and have constant backups

**Comment by Julusian** [comment]
@DachsbauTV you can do this if you set the `COMPANION_CONFIG_BASEDIR` environment variable when launching companion. I believe a shortcut can be made to launch it in this way.

**Comment by DachsbauTV** [comment]
Is there an explanation anywhere on how to to do this in win10? I've googled and looked through the repo but couldn't find anything unfortunately :/

I've tried adding it into the shortcut like this: "C:\Program Files\Companion\Companion.exe" /c set COMPANION_CONFIG_BASEDIR="C:\Users\DistilleryStation\OneDrive\Companion_Settings"

But now companion just shows "invalid" and doesn't work anymore.

**Comment by DachsbauTV** [comment]
And now it doesn't start at all anymore, even without the variable in the shortcut :/
Still just shows invalid

**Comment by Julusian** [comment]
The syntax is way worse than I was expecting it to be.. 
https://stackoverflow.com/a/34769146

**Comment by DachsbauTV** [comment]
Thanks for that. It started now with the settings, however, it still shows the undefined error and none of the buttons is clickable except for the dropdown menu

**Comment by maverickmueller** [comment]
In case anyone finds this later, the solution for me was to use just the directory name, relative to the executable. For instance, here is the batch file that I use to run it in this way:

SET COMPANION_CONFIG_BASEDIR=CompanionSettings
Companion.exe

This is with a folder created within the application directory called 'CompanionSettings'

**Comment by Fred-DTV** [comment]
amazing news @maverickmueller !

Just to clarify: Did you use a shortcut with the expression like this?
` "C:\Program Files\Companion\Companion.exe" /c set COMPANION_CONFIG_BASEDIR="CompanionSettings"`

**Comment by maverickmueller** [comment]
> amazing news @maverickmueller !
> 
> Just to clarify: Did you use a shortcut with the expression like this?
> `"C:\Program Files\Companion\Companion.exe" /c set COMPANION_CONFIG_BASEDIR="CompanionSettings"`
> 

I actually haven't tested using a shortcut. Just the batch job for now, and I'm not entirely sure how it would translate. But I assume so?

**Comment by Fred-DTV** [comment]
Could you maybe post your batch job?

Sorry I am a little over-cautious after last time I tried it scrambled my db's 🙈

**Comment by maverickmueller** [comment]
Shortcut wasn't working for me, since the COMPANION_CONFIG_BASEDIR value needs to be set before the executable is called, and isn't passed in as a parameter. But here's my batch setup:

My directory: 
![image](https://github.com/bitfocus/companion/assets/9595410/8873f386-ea8c-473d-81cd-e11bb275f4a4)

CompanionPortable.bat:
```
SET COMPANION_CONFIG_BASEDIR=config  
start "" Companion.exe
```
